aries_cloudagent/wallet/sd_jwt.py

```diff
@@ -0,0 +1,306 @@
+
+    return path_list
+
+
+def sort_sd_list(sd_list):
+    """
+    Sorts sd_list.
+
+    Ensures that selectively disclosable claims deepest
+    in the structure are handled first.
+    """
+    nested_claim_sort = [(len(sd.split(".")), sd) for sd in sd_list]
+    nested_claim_sort.sort(reverse=True)
+    return [sd[1] for sd in nested_claim_sort]
+
+
+def separate_list_splices(non_sd_list):
+    """
+    Separate list splices in the non_sd_list into individual indices.
+
+    This is necessary in order to properly construct the inverse of
+    the claims which should not be selectively disclosable in the case
+    of list splices.
+    """
+    for item in non_sd_list:
+        if ":" in item:
+            split = re.split(r"\[|\]|:", item)
+            for i in range(int(split[1]), int(split[2])):
+                non_sd_list.append(f"{split[0]}[{i}]")
+            non_sd_list.remove(item)
+
+    return non_sd_list
+
+
+def create_sd_list(payload, non_sd_list):
+    """Create a list of claims which will be selectively disclosable."""
+    flattened_payload = create_json_paths(payload)
+    separated_non_sd_list = separate_list_splices(non_sd_list)
+    sd_list = [
+        claim for claim in flattened_payload if claim not in separated_non_sd_list
+    ]
+    return sort_sd_list(sd_list)
+
+
+async def sd_jwt_sign(
+    profile: Profile,
+    headers: Mapping[str, Any],
+    payload: Mapping[str, Any],
+    non_sd_list: List = [],
+    did: Optional[str] = None,
+    verification_method: Optional[str] = None,
+) -> str:
+    """
+    Sign sd-jwt.
+
+    Use non_sd_list and json paths for payload elements to create a list of
+    claims that can be selectively disclosable. Use this list to wrap
+    selectively disclosable claims with SDObj within payload,
+    create SDJWTIssuerACAPy object, and call SDJWTIssuerACAPy.issue().
+    """
+    sd_list = create_sd_list(payload, non_sd_list)
+    for sd in sd_list:
+        jsonpath_expression = parse(f"$.{sd}")
+        matches = jsonpath_expression.find(payload)
+        if len(matches) < 1:
+            raise SDJWTError(f"Claim for {sd} not found in payload.")
+        else:
+            for match in matches:
+                if str(match.path) not in CLAIMS_NEVER_SD:
+                    if type(match.context.value) is list:
+                        match.context.value.remove(match.value)
+                        match.context.value.append(SDObj(match.value))
+                    else:
+                        match.context.value[
+                            SDObj(str(match.path))
+                        ] = match.context.value.pop(str(match.path))
+
+    sd_jwt_issuer = SDJWTIssuerACAPy(
+        user_claims=payload,
+        issuer_key=None,
+        holder_key=None,
+        profile=profile,
+        headers=headers,
+        did=did,
+        verification_method=verification_method,
+    )
+    await sd_jwt_issuer.issue()
+
+    return sd_jwt_issuer.sd_jwt_issuance
```

Comment:
Minor comment but since we've added the `issue()` method anyways, I think we should have the issue method return `self.sd_jwt_issuance` directly.